ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala

```diff
@@ -92,47 +92,51 @@ private[apiserver] final class ApiPartyManagementService private (
       .andThen(logger.logErrorsOnCall[ListKnownPartiesResponse])
   }
 
-  override def allocateParty(request: AllocatePartyRequest): Future[AllocatePartyResponse] =
-    withEnrichedLoggingContext(logging.partyString(request.partyIdHint)) {
-      implicit loggingContext =>
-        logger.info("Allocating party")
-        implicit val telemetryContext: TelemetryContext =
-          DefaultTelemetry.contextFromGrpcThreadLocalContext()
-        val validatedPartyIdHint =
-          if (request.partyIdHint.isEmpty) {
-            Future.successful(None)
-          } else {
-            Ref.Party
-              .fromString(request.partyIdHint)
-              .fold(
-                error =>
-                  Future.failed(
-                    ValidationLogger
-                      .logFailure(request, ValidationErrors.invalidArgument(error))
-                  ),
-                party => Future.successful(Some(party)),
-              )
-          }
-
-        validatedPartyIdHint
-          .flatMap(partyIdHint => {
-            val displayName = if (request.displayName.isEmpty) None else Some(request.displayName)
-            synchronousResponse
-              .submitAndWait(submissionIdGenerator(partyIdHint), (partyIdHint, displayName))
-              .map { case PartyEntry.AllocationAccepted(_, partyDetails) =>
-                AllocatePartyResponse(
-                  Some(
-                    PartyDetails(
-                      partyDetails.party,
-                      partyDetails.displayName.getOrElse(""),
-                      partyDetails.isLocal,
-                    )
+  override def allocateParty(request: AllocatePartyRequest): Future[AllocatePartyResponse] = {
+    val submissionId = submissionIdGenerator(request.partyIdHint)
```

Comment:
Can you also move the `contextualizedErrorLogger` in scope and pass this newly generated submissionId to it?

Reply:
That one is used in the `ValidationLogger.logFailure` below on line 114.

Reply:
ok